core/src/main/scala/filodb.core/query/QueryContext.scala

```diff
@@ -22,26 +22,50 @@ case class PromQlQueryParams(promQl: String, startSecs: Long, stepSecs: Long, en
 
 case object UnavailablePromQlQueryParams extends TsdbQueryParams
 
+case class IndividualQuota(
+        execPlanSamples: Int = 1000000,       // Limit on ExecPlan results in samples, default is 100K
+        execPlanResultBytes: Long = 18000000, // Limit on ExecPlan results in bytes, default is 18MB
+        groupByCardinality: Int = 100000,     // Limit on "group by" clause results, default is 100K
+        joinQueryCardinality: Int = 100000,   // Limit on binary join results, default is 100K
+        timeSeriesSamplesScanBytes: Long = 300000000)  // Limit on max data scanned per shard, default is 300 MB
+
+object IndividualQuota {
+  def apply(warnDefaults: Boolean): IndividualQuota = {
+    if (warnDefaults) {
+      IndividualQuota(
+        execPlanSamples = 50000,
+        execPlanResultBytes = 15000000,
+        groupByCardinality = 50000,
+        joinQueryCardinality = 50000,
+        timeSeriesSamplesScanBytes = 150000
+      )
+    } else {
+      IndividualQuota()
+    }
+  }
+}
 case class PlannerParams(applicationId: String = "filodb",
                          spread: Option[Int] = None,
                          spreadOverride: Option[SpreadProvider] = None,
                          shardOverrides: Option[Seq[Int]] = None,
                          targetSchemaProviderOverride: Option[TargetSchemaProvider] = None,
                          queryTimeoutMillis: Int = 60000, // set default to match default http-request-timeout
-                         sampleLimit: Int = 1000000,
-                         groupByCardLimit: Int = 100000,
-                         joinQueryCardLimit: Int = 100000,
-                         resultByteLimit: Long = 18000000,  // 18MB
+                         enforcedQuota: IndividualQuota = IndividualQuota(),
+                         warnQuota: IndividualQuota = IndividualQuota.apply(warnDefaults = true),
```

Comment:
pls see comment above, will update the description in the PR as well.
In short the warn most likely in the first phase will go to logs. It would be used by operators to see what the reasonable limits are.